README.md

```diff
@@ -110,6 +110,17 @@ You can use an in-memory SQLite driver by adding the `-M` option.
 You can choose to include helper files. This is not enabled by default, but you can override it with the `--helpers (-H)` option.
 The `Illuminate/Support/helpers.php` is already set up, but you can add/remove your own files in the config file.
 
+### Automatic PHPDoc generation for macros
+
+The generator can detect macros and mixins which are declared in any loaded service provider.
+A basic autocompletion with parameter and return types can be generated if you use type hinting:
+
+```php
+Str::macro('concat', function(string $str1, string $str2) : string {
+    return $str1 . $str2;
+});
```

Comment:
When I try this, it doesn't autocomplete:
![image](https://user-images.githubusercontent.com/87493/90819113-d3868d00-e32f-11ea-9728-6f863f804c6c.png)

Because the method isn't declared as `static`, here's the generated stub from `_ide_helper.php`:
```php
    namespace Illuminate\Support {
    
        class Str {
        
            /**
         * No description available.
         *
         * @param string $str1
         * @param string $str2
         * @return string 
         */
            public function concat($str1, $str2) { }
        
        }
    
    }
```
When I manually change it, it works:
![image](https://user-images.githubusercontent.com/87493/90819309-19435580-e330-11ea-8b97-adecd2e74d31.png)

Without the `static`, you _can_ hover over it and it shows you the non-static callable version.

I guess it's "either" or "or": static or not, because the `Macroable` trait itself simply looks up either way it's called and doesn't really care.

Would it work if instead of generating the stub with PHP code methods, we would generated something like this:
```php
    namespace Illuminate\Support {

        /**
         * @method static string concat(string $str1, $str2)
         * @method string concat(string $str1, $str2)
         */
        class Str {
        }

    }
```
🤔

Reply:
I think I have solved the problem by just adding the static tag. PhpStorm should now recognize the macros in static and non-static contexts.